src/Drips.sol

```diff
@@ -286,11 +286,14 @@ library Drips {
             require(receiver.amtPerSec != 0, "Drips receiver amtPerSec is zero");
             if (i > 0) require(_isOrdered(receivers[i - 1], receiver), "Receivers not sorted");
             uint64 start = receiver.start;
-            if (start < dripsStart) start = dripsStart;
+            if (receiver.start == 0) start = dripsStart;
             if (receiver.duration == 0) {
+                if (start < dripsStart) start = dripsStart;
                 length = _addDefaultEnd(defaults, length, start, receiver.amtPerSec);
             } else {
                 uint64 end = _capTimestamp(uint72(start) + receiver.duration);
```

Comment:
I think it makes sense to calculate the end time based on the user passed start time.

Reply:
On another branch I've refactored the code calculating the actual drips range to be a commonly used function instead of multiple copies of the same logic. I think that this will solve this readability issue in a more elegant way.

Reply:
Ah okay, cool.